src/go/configmanager/flags/flags.go

```diff
@@ -160,7 +160,14 @@ var (
         x-envoy-retry-grpc-on(https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter#x-envoy-retry-on).`)
 	BackendRetryNum = flag.Uint("backend_retry_num", 1,
 		`The allowed number of retries. Must be >= 0 and defaults to 1. This retry
-	setting will be applied to all the backends if you have multiple ones.`)
+        setting will be applied to all the backends if you have multiple ones.`)
+	BackendPerTryTimeoutSec = flag.Uint("backend_per_try_timeout_sec", 0,
```

Comment:
Nit: change to flag.Duration and have python pass as string?

Reply:
We have retry setting by default. To the contrary, users will be surprised when the real deadline=deadline * (1 + retry_num) if we use per_try_timeout=deadline without knowing retry setting. Potentially, it is a breaking change. If we set `per_try_timeout` here, it would be transparent to users. If users would like to set the retry in the flags, they can see the description about the relationship between x-google-backend.deadline and per_try_timeout. 

I regard it as part of retry settings so it is more natural to put them together in deployment flags.
 
As for the per-route retry, actually we have same discussion when we were adding the retry. We doubted per route retry would be really needed for our customers, so we added other retry setting as flag.